Sources/Core/Shared/ConditionOperation.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  ConditionOperation.swift
+//  Operations
+//
+//  Created by Daniel Thorpe on 15/04/2016.
+//
+//
+
+import Foundation
+
+
+/**
+ Condition Operation
+
+ Conditions are a core feature of this framework. Multiple
+ instances can be attached to an `Operation` subclass, whereby they
+ are evaluated to determine whether or not the target operation is
+ executed.
+
+ ConditionOperation is also an Operation subclass, which means that it
+ also benefits from all the features of Operation, namely dependencies,
+ observers, and yes, conditions. This means that your conditions could
+ have conditions. This allows for expressing incredibly rich control logic.
+
+ Additionally, conditions are evaluated asynchronously, and indicate
+ failure by passing an ConditionResult enum back.
+
+ */
+public class ConditionOperation: Operation {
+    public typealias CompletionBlockType = ConditionResult -> Void
+
+    public var isMutuallyExclusive: Bool = false
+
+    internal weak var operation: Operation? = .None
```

Comment:
Technically I think this should be `unowned` but Swift won't allow me to use it.